packages/create-plugin/templates/app/src/components/AppConfig/AppConfig.tsx

```diff
@@ -44,8 +46,30 @@ export const AppConfig = ({ plugin }: AppConfigProps) => {
     });
   };
 
+  const onSubmit = () => {
+    if (isSubmitDisabled) {
+      return;
+    }
+
+    updatePluginAndReload(plugin.meta.id, {
+      enabled,
+      pinned,
+      jsonData: {
+        apiUrl: state.apiUrl,
+      },
+      // This cannot be queried later by the frontend.
+      // We don't want to override it in case it was set previously and left untouched now.
+      secureJsonData: state.isApiKeySet
+        ? undefined
+        : {
+            apiKey: state.apiKey,
+          },
+    });
+  };
+
   return (
     <div data-testid={testIds.appConfig.container}>
+    <form onSubmit={onSubmit}>
```

Comment:
nit: could put the data-testid on the form and remove the div wrapper? 🤔
if not keen on my other nit is the form element needs indenting. 😄

Reply:
Good point, will do! :)

Reply:
Doesn't look like the testIds in this form are being used anywhere in this app. Probably best to remove them entirely (but fine with me to do that in a separate PR too).

Reply:
Hmm, yeah it probably makes sense to remove them then. 👍

Reply:
Addressed in https://github.com/grafana/plugin-tools/pull/1041/commits/1ab47f9435853af26327c194ec7eece7b6ad533c.